.github/workflows/rust.yml

```
@@ -7,11 +7,18 @@ on:
     branches: [ main ]
 
 jobs:
-  build:
+  build-and-deploy:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v2
     - name: Build
       run: ./containerized-build.sh build
     - name: Lint
       run: ./containerized-build.sh lint
+    - name: Build book
+      run: ./scripts/mdbook.sh
+    - name: Deploy website
+      uses: JamesIves/github-pages-deploy-action@4.1.4
+      with:
+        branch: gh-pages
+        folder: docs/book
```

Keeping this job in this file might cause one problem.

At the top we have:

```
on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]
```

Which means that every step defined in this file is running:

* when a commit is pushed to the `main` branch - which happens when a PR is merged - I think that's exactly when we want to deploy docs :)
* when a PR is submitted - and in this case, I think that we **don't** want to deploy the website

So I think that we should do two things.

First one, keep only building the docs in this file, without deploying:

```suggestion
    - name: Build book
      run: ./scripts/mdbook.sh
```

And then create a separate file, i.e. `.github/workflows/mdbook.yml`, with the following content:

```
name: mdBook

on:
  push:
    branches: [ main ]

jobs:
  build-and-deploy:
    runs-on: ubuntu-latest
    steps:
    - name: Build book
      run: ./scripts/mdbook.sh
    - name: Deploy website
      uses: JamesIves/github-pages-deploy-action@4.1.4
      with:
        branch: gh-pages
        folder: docs/book
```

I think that should work. And also make this pull request green. Could you try it?

---

yes, it will be more reasonable solution